src/libs/nav.ts

```diff
@@ -7,6 +7,38 @@ import { useOnMount, useStore } from 'src/libs/react-utils';
 import { routeHandlersStore } from 'src/libs/state';
 import * as Utils from 'src/libs/utils';
 
+// TODO: add all used nav key names here and switch consumers to use the safe const values here
+/**
+ * union type (enum-like) of safe nav key names
+ */
+export type TerraNavKey = 'billing' | 'workspace-dashboard' | 'workspaces';
+
+export type TerraNavKeyLookup = { [key in TerraNavKey]: key };
+
+// TODO: fill out this value lookup to reflect full list of key names once available above
+/**
+ * allowed values for nav keys in a lookup table
+ * Note: use terraNavKey type guard method below instead of this const.
+ *   Typescript currently allows unsafe object['not-a-enum-key-name'] indexing
+ *   lookups into this otherwise type-safe construct.
+ */
+export const terraNavKeyValue: TerraNavKeyLookup = {
+  billing: 'billing',
+  'workspace-dashboard': 'workspace-dashboard',
+  workspaces: 'workspaces',
+};
```

Comment:
I would construct this differently to make the intent clearer and reduce the possibility of typos:
```suggestion
export const terraNavKeyValue: TerraNavKeyLookup = [
  'billing', 'workspace-dashboard', 'workspaces'
].reduce((acc, k) => ({...acc, [k]: k}), {});
```

Reply:
love it!

Reply:
actually, I misread the suggestion... missed the reduce at the end... Typescript will prevent typos already (TerraNavKeyLookup insures that only valid keys can be used, and that the value must match the key).   While this suggestion was attractive because it removed the key: key repetition in the main body of the object, the reduce mechanic causes new Typescript problems since it's a bit too dynamic to guarantee the end result of .reduce is the full set of keys.  I got TS complaints when I tried this locally, and the type protection seemed to actually suffer in general.

Reply:
I can find some ways of doing it generically, but they're all way more complex and opaque than just writing them out.